src/shared/calendarspec.c

```diff
@@ -1345,6 +1372,16 @@ static int find_next(const CalendarSpec *spec, struct tm *tm, usec_t *usec) {
                 if (r == 0)
                         continue;
 
+                r = tm_compare(tm, &c);
+                if (r == 0)
+                        r = CMP(*usec - 1, (usec_t) tm_usec);
```

Comment:
hmm, can't this underflow? i.e. if *usec is zero? 

I think it would make more sense to add 1 to tm_usec instead, since we know that that's not going to be UINT64_MAX since we treat that as USEC_INFINITY, which we don't allow.

(maybe add an `assert()` here checking for this)

Reply:
asserts to enforce against programming errors are common and used all over the place so it's ok